test/e2e/tests/settings.spec.ts

```diff
@@ -0,0 +1,304 @@
+    // navigate to main settings URL and verify general settings displayed by default
+    await settingsPage.gotoMainSettingsPage();
+    await expect(settingsPage.settingsHeaderEN).toBeVisible({ timeout: TIMEOUT_30_SECONDS }); // generous time
+    await expect(settingsPage.generalSettingsHeaderEN).toBeVisible({ timeout: TIMEOUT_30_SECONDS }); // generous time
+    await expect(settingsPage.generalSettingsBtn).toBeEnabled();
+    await expect(settingsPage.calendarSettingsBtn).toBeEnabled();
+    await expect(settingsPage.accountSettingsBtn).toBeEnabled();
+    await expect(settingsPage.connectedSettingsBtn).toBeEnabled();
+
+    // click 'Calendar' button and verify corresponding settings appear
+    await settingsPage.calendarSettingsBtn.click();
+    await expect(settingsPage.calendarSettingsHeader).toBeVisible({ timeout: TIMEOUT_30_SECONDS }); // generous time
+
+    // click 'Account' button and verify corresponding settings appear
+    await settingsPage.accountSettingsBtn.click();
+    await expect(settingsPage.accountSettingsHeader).toBeVisible({ timeout: TIMEOUT_30_SECONDS }); // generous time
+
+    // click 'Connected Accounts' button and verify corresponding settings appear
+    await settingsPage.connectedSettingsBtn.click();
+    await expect(settingsPage.connectedSettingsHeader).toBeVisible({ timeout: TIMEOUT_30_SECONDS }); // generous time
+
+    // click 'General' button and verify general settings appear once again
+    await settingsPage.generalSettingsBtn.click();
+    await expect(settingsPage.generalSettingsHeaderEN).toBeVisible({ timeout: TIMEOUT_30_SECONDS }); // generous time
+  });
+});
+
+test.describe('general settings', {
+  tag: [PLAYWRIGHT_TAG_E2E_SUITE],
+}, () => {
+  test.beforeEach(async ({ page }) => {
+    // navigate to and sign into appointment
+    await navigateToAppointmentAndSignIn(page);
+    settingsPage = new SettingsPage(page);
+    dashboardPage = new DashboardPage(page);
+    // navigate to the general settings page
+    await settingsPage.gotoGeneralSettingsPage();
+  });
+
+  test('able to change language', async ({ page }) => {
+    // change language setting to DE and verify; we use expect.soft here because if the expect fails
+    // we still want the test to continue so that the test will change the setting back to original value
+    // (expect.soft will still mark the test case failed if the expect fails, but won't stop the test)
+    await settingsPage.changeLanguageSetting(APPT_LANGUAGE_SETTING_DE);
+    await expect.soft(settingsPage.settingsHeaderDE).toBeVisible({ timeout: TIMEOUT_30_SECONDS });
+    await expect.soft(settingsPage.generalSettingsHeaderDE).toBeVisible();
+
+    // change language settings back to EN and verify
+    await settingsPage.changeLanguageSetting(APPT_LANGUAGE_SETTING_EN);
+    await expect(settingsPage.settingsHeaderEN).toBeVisible({ timeout: TIMEOUT_30_SECONDS });
+    await expect(settingsPage.generalSettingsHeaderEN).toBeVisible();
```

Comment:
Can you check local storage in playwright? It might be a good idea to ensure the changes there have persisted to it.